analyzer/codechecker_analyzer/analyzers/clangsa/version.py

```diff
@@ -46,17 +48,22 @@ def __init__(self):
     def parse(self, version_string):
         """Try to parse the version string using the predefined patterns."""
         version_match = re.search(self.clang_version_pattern, version_string)
+        if not version_match:
+            return
+
         installed_dir_match = re.search(
             self.clang_installed_dir_pattern, version_string)
 
-        if not version_match or not installed_dir_match:
-            return False
+        if installed_dir_match:
+            installed_dir = installed_dir_match.group('installed_dir')
+        else:
+            installed_dir = os.path.dirname(shutil.which(self.analyzer_binary))
```

Comment:
However, it is quite unlikely but having `version_string` parameter and `self.analyzer_binary` belonging to different `clang` binaries, this function can return a `ClangVersionInfo` object which contains mixed information of two clangs. Let's keep it in mind in case it occurs sometime, but it looks good apart from this.